src/display/editor/annotation_editor_layer.js

```diff
@@ -783,6 +793,13 @@ class AnnotationEditorLayer {
     this.#linkNodeParams = null;
   }
 
+  createAnnotationNode() {
+    this.removeTempHighlight();
+    const { event, isCentered, ...highlightParams } = this.#highlightNodeParams;
+    this.#createAndAddNewEditor(event, isCentered, highlightParams);
+    this.#highlightNodeParams = null;
```

Comment:
Remove this line and move it to `removeTempHighlight` where link node params are reset. `createAnnotationNode` is only called when highlight has to be made whereas `removeTempHighlight` is always called, which guarantees that `#highlightSelectionParams` will never have invalid value